### PR TITLE
Bug fix: remapping of CatchCN restart variables (RZMM, TGWM, SFMM)

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
Updated 4 May 2023 by @gmao-rreichle: PR fixes a bug in remapping of CatchCN restarts.  Some CatchCN restart variables (RZMM, TGWM, SFMM) ended up with wrong array dimensions and were set to zero (rather than regridded).  See also #658, which fixed a similar bug for other variables but missed RZMM, TGWM, and SFMM.
PR is trivially 0-diff for all Catch simulations (LDAS and GCM) and also for CatchCN simulations that do not rely on regridding of restarts during setup.  Non-0-diff for regridding of CatchCN restarts.
---
@gmao-jkolassa , This PR should fix your regridding error.  I will remove the draft label after you confirm.

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
This fix has been tested. @sdrabenh

**Comment by gmao-rreichle** [comment]
Converting to draft PR pending review by surface-preproc-team.
@weiyuan-jiang, please initiate PRs as "draft" and convert to "ready for review" only after full testing and approval by the relevant development team(s).  Then @sdrabenh will know to ignore the PR until we've fully signed off on it.

**Comment by gmao-jkolassa** [comment]
@weiyuan-jiang : Upon further review today it seems there is another issue. The variables `rzmm`, `tgwm` and `sfmm` should all be regridded instead of being set to 0. It appears that when we fixed the regridding of the CatchCN variables (when we changed it so they are actually regridded instead of all being set to 0; PR #658 ), these were missed. I will make the appropriate changes to `CatchmentCNRst.F90` to include them and then ask you to review those.

**Comment by biljanaorescanin** [comment]
@gmao-jkolassa will this affect restarts you gave to @mathomp4 for CatchCN Nightly Tests?

**Comment by gmao-jkolassa** [comment]
What Matt currently has is zero-diff with respect to what is currently run. The suggested change is a bug fix, which once implemented will be not zero-diff with respect to the current version of the model.

**Comment by mathomp4** [comment]
> What Matt currently has is zero-diff with respect to what is currently run. The suggested change is a bug fix, which once implemented will be not zero-diff with respect to the current version of the model _if_ restart files are being regridded.

That said, if you'd like to make new restarts with the fix and pass them along, I can update mine and re-baseline.

Indeed, it's probably good for me to have "good" restarts around in case the ones I have have issues.

**Comment by gmao-jkolassa** [comment]
Sure. I am working on implementing this today and if all looks good I should be able to pass you the new restarts later today or tomorrow.

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CatchmentCNRst.F90 [hunk 1]
```diff
@@ -817,6 +817,16 @@ subroutine re_tile(this, InTileFile, OutBcsDir, OutTileFile, surflay, rc)
         enddo
         this%psnsham = var_psn
 
+        do nz = 1, nzone
+           var_out_zone(:,nv) = this%rzmm(this%id_glb(:), nv)
```
Comment (weiyuan-jiang) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CatchmentCNRst.F90:822]: The index nv probably should be nz

Comment (gmao-jkolassa) [reply GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CatchmentCNRst.F90]: Good catch! Fixed it in latest commit.

**Comment by gmao-jkolassa** [comment]
This is just to note that I have tested this for Catchment-CN4.0 and Catchment-CN4.5 and `rzmm`, `sfmm` and `tgwm` are now being regridded. From my end this PR is good to go.

**Comment by gmao-rreichle** [comment]
Thanks, @gmao-jkolassa, for the changes and for asserting that the fix works. 
Since the PR only touches `CatchmentCNRst.F90`, it is trivially 0-diff for all Catch simulations (LDAS and GCM) and also for CatchCN simulations that do not rely on regridding of restarts during setup.
I assume that CatchCN simulations that do rely on regridding of restarts are **not** 0-diff and attached the "non-0-diff" label in addition to the "0-diff" label.  
I don't think that further testing is needed, will approve for the surface-preproc-team, and will forward to @sdrabenh for his review and merging.
cc: @biljanaorescanin @weiyuan-jiang

**Review by gmao-rreichle — APPROVED** [review]

**Comment by gmao-rreichle** [comment]
@sdrabenh: The "build-test" succeeds in building the model but fails in the run.  The error message points to "SS2G_GridCompMod.F90" and doesn't seem related to the PR.  Not sure what the problem might be. 
cc: @biljanaorescanin @weiyuan-jiang @gmao-jkolassa

**Review by sdrabenh — APPROVED** [review]

**Pull request merged**